src/index.d.ts

```diff
@@ -120,647 +156,1053 @@ declare namespace connect {
+     *
+     * @param callbacks Success and failure callbacks to determine whether the operation was successful.
      */
-    resume(successFailOptions: SuccessFailOptions): void;
-    getMonitorInfo(): MonitorInfo;
+    resume(callbacks?: SuccessFailOptions): void;
   }
 
-  interface VoiceConnection extends BaseConnection {
-    getMediaType(): MediaType;
-    getMediaInfo(): any;
-    getMediaController(): any;
-    sendDigits(digits: string, options: SuccessFailOptions);
-    hold(options: SuccessFailOptions);
-    resume(options: SuccessFailOptions);
-    isOnHold(): boolean;
-  }
+  /**
+   * The VoiceConnection API provides action methods (no event subscriptions) which can be called to manipulate the state of a particular voice connection within a contact.
+   * Like contacts, connections come and go.
+   * It is good practice not to persist these object or keep them as internal state.
+   * If you need to, store the `contactId` and `connectionId` of the connection and make sure that the contact and connection still exist by fetching them in order from the `Agent` API object before calling methods on them.
+   */
+  class VoiceConnection extends BaseConnection {
+    /** Returns the media info object associated with this connection. */
+    getMediaInfo(): VoiceMediaInfo;
+
+    /** Returns the `MediaType` enum value: `"softphone"`. */
+    getMediaType(): MediaType.SOFTPHONE;
 
-  interface ChatConnection extends BaseConnection {
-    getMediaType(): MediaType;
-    getMediaInfo(): any;
-    getMediaController(): any;
-    getConnectionToken(): Promise<{ chatTokenTransport: { participantToken: string } }>
+    /** Get the media controller associated with this connection. */
+    getMediaController(): Promise<VoiceMediaInfo>;
   }
 
-  type Connection = ChatConnection | VoiceConnection;
+  /**
+   * The ChatConnection API provides action methods (no event subscriptions) which can be called to manipulate the state of a particular chat connection within a contact.
+   * Like contacts, connections come and go.
+   * It is good practice not to persist these object or keep them as internal state.
+   * If you need to, store the `contactId` and `connectionId` of the connection and make sure that the contact and connection still exist by fetching them in order from the `Agent` API object before calling methods on them.
+   */
+  class ChatConnection extends BaseConnection {
+    /** Get the media info object associated with this connection. */
+    getMediaInfo(): ChatMediaInfo;
+
+    /** Provides a promise which resolves with the API response from createTransport transportType chat_token for this connection. */
+    getConnectionToken(): Promise<ConnectionToken>;
+
+    /** Returns the `MediaType` enum value: `"chat"`.  */
+    getMediaType(): MediaType.CHAT;
+
+    /** Get the media controller associated with this connection. */
+    getMediaController(): Promise<any>;
```

Comment:
should this be `Promise<ChatMediaInfo>`?

Reply:
That was my first thought yes. However, this happened on my tests:

```ts
// this threw an Error for chat contacts unless I included
// https://github.com/amazon-connect/amazon-connect-chatjs
const cnn = new connect.Agent().getContacts()[0].getConnections()[0];

// I was never able to get an instance of this (could not verify it).
// Hence, I stuck with the details on `Documentation.md`
cnn.getMonitorInfo() === null; // true

if (cnn instanceof connect.VoiceConnection) {
  const mediaInfo = cnn.getMediaInfo();
  cnn.getMediaController().then(mediaController => {
    // This was true, hence I defined `getMediaController(): Promise<VoiceMediaInfo>;`
    // rather than `getMediaController(): Promise<SoftphoneMediaController>;`
    JSON.stringify(mediaInfo) === JSON.stringify(mediaController);

    // This always yielded false, which is counter-intuitive with
    // https://github.com/amazon-connect/amazon-connect-streams/blob/master/src/mediaControllers/factory.js#L46
    mediaController instanceof connect.SoftphoneMediaController;
  });
} else if (cnn instanceof connect.ChatConnection) {
  const mediaInfo = cnn.getMediaInfo();
  cnn.getMediaController().then(mediaController => {
    // This was false, and the data definition was nowhere to be found on the package
    // I concluded that this must come from https://github.com/amazon-connect/amazon-connect-chatjs
    // hence I defined `getMediaController(): Promise<any>;`
    JSON.stringify(mediaController) === JSON.stringify(mediaInfo); // false

    // This always yielded false, which is counter-intuitive with
    // https://github.com/amazon-connect/amazon-connect-streams/blob/master/src/mediaControllers/factory.js#L44
    mediaController instanceof connect.ChatMediaController;
  })
}
```

---

I actually wonder whether the `getMonitorInfo()`, `getMediaInfo()`, and `getMediaController()` methods should be part of the public API.

After reading the documentation, I have no idea what they're for, and it seemed that they only contain internally used data.

Reply:
@ctwomblyamzn any inputs on this?

Reply:
As of now, the voice mediaController is a placeholder. The chat mediaController is ChatJS' chatController. That is definitely intended to be a public API. Voice functionality has not yet been transitioned from softphone.js to a mediaController model. Right now, we are not sure when that will happen. I think the best solution here is to update the docs to indicate that `VoiceConnection.getMediaController()` is a placeholder for now, but that `ChatConnection.getMediaController()` is functional, and that it returns a chatSession object, referred to as the chat media controller. More info about the chatSession object is available in the ChatJS documentation: https://github.com/amazon-connect/amazon-connect-chatjs.

Reply:
Given this information, I have opted for:

- Update the documentation.
- Have both `VoiceConnection.getMediaController()` and `ChatConnection.getMediaController()` return a `Promise<any>`.